### DISCUSSION TITLE
Installing streamlink on Libreelec

### DISCUSSION THREAD

**Discussion by ccontrolled** [streamlink/streamlink#4958] [Q&A] [answered]
### Streamlink version

Latest stable release

### Description

Is there any Addon or help how to install streamlink on Libreelec in a simple way?

### Debug log

```text
2022-11-12 16:49:18.921 T:10044    INFO <general>: Loading skin file: FileBrowser.xml, load type: KEEP_IN_MEMORY
2022-11-12 16:49:19.755 T:6328     INFO <general>: CActiveAESink::OpenSink - initialize sink
2022-11-12 16:49:30.794 T:10044   ERROR <general>: CAddonInfoBuilder::ADDON::CAddonInfoBuilder::Generate: Unable to load 'zip://C%3a%5cUsers%5ch%5cDocuments%5cstreamlink-5.0.1.zip/streamlink-5.0.1/addon.xml', Line 0
                                                   Failed to open file
```

**Comment by mkbloke** [comment, marked as answer]
You should probably ask on the Libreelec or Kodi forums.  I'm not sure anybody here can really help.

~You can try starting with the info in this somewhat old [thread](https://forum.libreelec.tv/thread/14643-adding-steamlink-to-v9-0-0/).~

~The last post from the OP of the thread on [Feb 25th 2022](https://forum.libreelec.tv/thread/14643-adding-steamlink-to-v9-0-0/?postID=167126#post167126) might be useful as that contains an updated script.~

**Reply by mkbloke** [reply]
I've just noticed the lack of an `r` in the above thread - it's for something else (`steamlink` not `streamlink`), so it's not helpful.

**Reply by mkbloke** [reply]
Maybe an AppImage will work?  I don't really know as I don't use Libreelec.  You'll have to try and work it out for yourself.  Looks like you'll need aarch64 for an RPi 3/4 otherwise likely the x86_64 version depending on the architecture of your device.  Good luck.

https://github.com/streamlink/streamlink-appimage/releases/tag/5.1.0-1

**Comment by cubimol** [comment]
As far as I know, streamlink cannot be installed directly in LibreELEC, but it can be installed alongside tvheadend. To do this, install docker linuxserver/tvheadend and prepare the container to run at start the script https://github.com/cgomesu/tvhlink/blob/master/tools/docker/streamlink_for_tvh_container.sh created by Carlos Gomes. It works very well.

**Comment by ccontrolled** [comment]
Why is it impossible to make an addon for Libreelec?
FFMPEG - addon works well on Libreelec with a path to ffmpeg.
But script.module.streamlink-2.3.0+matrix.4.zip from jairoxyz:
-sh: streamlink: not found  :(

**Reply by mkbloke** [reply]
Nobody is saying it's impossible.  It just requires somebody with the right knowledge (or willing to gain that knowledge) and the motivation to do it.
